### PR TITLE
[#1724, #1741, #1805] Fix up focus trap for dialogs

### PR DESCRIPTION
**Pull request by corinagum**
#1724, #1741, #1805 

Before undrafting this issue: 
- [x] merge/pull #1828 
- [x] merge/pull #1763 
- [x] add tests

This fixes the focus trap on dialogs.

### CODE REVIEW

**Comment by tonyanziano** [comment]
#1805 is related as well

**Comment by corinagum** [comment]
working on the tests now.

**Comment by corinagum** [comment]
This is ready for review. :)

File: packages/app/main/e2e/navigateADialog.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    expect(await app.client.hasFocus("a*=Don't have an Azure Account?")).toBeTruthy();
+
+    // press tab to move focus within the dialog
+    await app.client.keys(['\uE004', '\uE004', '\uE004']);
+    expect(await app.client.hasFocus('button=Sign in with Azure')).toBeTruthy();
+
+    // press tab again to circle through the focus trap in the dialog
+    await app.client.keys(['\uE004', '\uE004']);
+    expect(await app.client.hasFocus("a*=Don't have an Azure Account?")).toBeTruthy();
+
```
Comment (tonyanziano) [inline packages/app/main/e2e/navigateADialog.spec.ts]: nice!

File: packages/app/main/e2e/navigateADialog.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+
+import { existsSync, unlinkSync } from 'fs';
+import { Application } from 'spectron';
+
+import { appPath, chromeDriverLogPath, electronPath } from './utils';
+
+import { init, setupMock } from './mocks/electronDialog';
```
Comment (tonyanziano) [inline packages/app/main/e2e/navigateADialog.spec.ts]: `setupMock` is unused

File: packages/app/main/e2e/navigateADialog.spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+//
+// THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+
+import { existsSync, unlinkSync } from 'fs';
```
Comment (tonyanziano) [inline packages/app/main/e2e/navigateADialog.spec.ts]: unused

**Comment by corinagum** [comment]
@tonyanziano ping, build/tests are passing. thanks for your patience

**Pull request merged**